core/src/lib.rs

```diff
@@ -174,3 +179,66 @@ mod sealed {
         }
     }
 }
+
+pub struct CoreRuntime {
```

Comment:
This abstraction is great. Brings us one step closer to `new_remote` that would connect to an external/WASM'd core.

For TS and Python (and .Net and Ruby), I recommend we have an in-language `Runtime` concept backed by this, w/ a lazily loaded singleton. Then each client (and worker if it's separate from client) can optionally accept a _different_ runtime or leave unset to use the singleton.

Reply:
Yeah, I agree. In fact, I am adding something like that for my tests just so that I don't unnecessarily spawn a bajillion telemetry instances while they run